pkg/svc/extension.go

```diff
@@ -406,82 +413,64 @@ func (s *extensionRegistrySvc) DeleteService(ctx context.Context, req *extension
 // extension registry.
 func (s *extensionRegistrySvc) AddEndpoint(ctx context.Context, req *extension.ExtensionEndpoint) (res *extension.ExtensionEndpoint, err error) {
 	s.logger.Print("extension.addEndpoint")
-	endpoint, err := s.registry.AddEndpoint(ctx, extensionEndpointToDomain(req))
+	endpoint, err := s.registry.AddEndpoint(ctx, *req.ExtensionID, *req.ServiceID, extensionEndpointToDomain(req))
 	if err != nil {
 		return nil, errToRest(err)
 	}
-	return extensionEndpointToRest(endpoint), nil
+	return extensionEndpointToRest(ctx, *req.ExtensionID, *req.ServiceID, endpoint), nil
 }
 
 // Retrieve information about an endpoint belonging to an extension.
 func (s *extensionRegistrySvc) GetEndpoint(ctx context.Context, req *extension.GetEndpointPayload) (res *extension.ExtensionEndpoint, err error) {
 	s.logger.Print("extension.getEndpoint")
-	endpoint, err := s.registry.GetEndpoint(ctx, domain.ExtensionEndpointID{
-		ExtensionID: req.ExtensionID,
-		ServiceID:   req.ServiceID,
-		URL:         extensionEndpointURLToDomain(&req.URL),
-	})
+	endpoint, err := s.registry.GetEndpoint(ctx, req.ExtensionID, req.ServiceID, extensionEndpointURLToDomain(&req.URL))
 	if err != nil {
 		return nil, errToRest(err)
 	}
-	return extensionEndpointToRest(endpoint), nil
+	return extensionEndpointToRest(ctx, req.ExtensionID, req.ServiceID, endpoint), nil
 }
 
 // List all endpoints associated with an extension service registered in FuseML
 func (s *extensionRegistrySvc) ListEndpoints(ctx context.Context, req *extension.ListEndpointsPayload) (res []*extension.ExtensionEndpoint, err error) {
 	s.logger.Print("extension.listEndpoints")
-	svcRecord, err := s.registry.GetService(ctx, domain.ExtensionServiceID{
-		ExtensionID: req.ExtensionID,
-		ID:          req.ServiceID,
-	}, true)
+	svc, err := s.registry.GetService(ctx, req.ExtensionID, req.ServiceID)
 	if err != nil {
 		return nil, errToRest(err)
 	}
-	res = make([]*extension.ExtensionEndpoint, len(svcRecord.Endpoints))
-	for i, endpoint := range svcRecord.Endpoints {
-		res[i] = extensionEndpointToRest(endpoint)
-	}
+	res = extensionEndpointsToRest(ctx, req.ExtensionID, req.ServiceID, svc.ListEndpoints(ctx))
 	return res, nil
 }
 
 // Update an endpoint belonging to an extension service registered in FuseML
 func (s *extensionRegistrySvc) UpdateEndpoint(ctx context.Context, req *extension.ExtensionEndpoint) (res *extension.ExtensionEndpoint, err error) {
 	s.logger.Print("extension.updateEndpoint")
 	endpoint := extensionEndpointToDomain(req)
-	ep, err := s.registry.GetEndpoint(ctx, domain.ExtensionEndpointID{
-		URL:         extensionEndpointURLToDomain(&endpoint.URL),
-		ExtensionID: endpoint.ExtensionID,
-		ServiceID:   endpoint.ServiceID,
-	})
+	ep, err := s.registry.GetEndpoint(ctx, *req.ExtensionID, *req.ServiceID, extensionEndpointURLToDomain(&endpoint.URL))
```

Comment:
what if the extension ID or the service ID are nil ?

Reply:
same here